diff -up audacious-plugins-2.2/src/ffaudio/ffaudio-core.c~ audacious-plugins-2.2/src/ffaudio/ffaudio-core.c
--- audacious-plugins-2.2/src/ffaudio/ffaudio-core.c~	2009-11-22 23:49:52.000000000 +0100
+++ audacious-plugins-2.2/src/ffaudio/ffaudio-core.c	2009-12-19 15:15:03.000000000 +0100
@@ -664,6 +664,9 @@ static gchar *ffaudio_fmts[] = {
     /* VQF */
     "vqf",
 
+    /* Apple Lossless (alac) */
+    "m4a",
+
     /* end of table */
     NULL
 };
