diff -up audacious-plugins-2.4-rc2/src/mpg123/Makefile~ audacious-plugins-2.4-rc2/src/mpg123/Makefile
--- audacious-plugins-2.4-rc2/src/mpg123/Makefile~	2010-08-13 23:01:12.000000000 +0200
+++ audacious-plugins-2.4-rc2/src/mpg123/Makefile	2010-08-24 11:58:06.147712997 +0200
@@ -1,27 +1,7 @@
 PLUGIN = madplug${PLUGIN_SUFFIX}
 
 SRCS = mpg123.c \
-       id3skip.c \
-       libmpg123/compat.c \
-       libmpg123/dct64.c \
-       libmpg123/equalizer.c \
-       libmpg123/feature.c \
-       libmpg123/format.c \
-       libmpg123/frame.c \
-       libmpg123/icy2utf8.c \
-       libmpg123/icy.c \
-       libmpg123/id3.c \
-       libmpg123/index.c \
-       libmpg123/layer1.c \
-       libmpg123/layer2.c \
-       libmpg123/layer3.c \
-       libmpg123/libmpg123.c \
-       libmpg123/optimize.c \
-       libmpg123/parse.c \
-       libmpg123/readers.c \
-       libmpg123/stringbuf.c \
-       libmpg123/synth.c \
-       libmpg123/tabinit.c
+       id3skip.c
 
 include ../../buildsys.mk
 include ../../extra.mk
@@ -31,4 +11,4 @@ plugindir := ${plugindir}/${INPUT_PLUGIN
 CFLAGS += ${PLUGIN_CFLAGS}
 CPPFLAGS += ${PLUGIN_CPPFLAGS} ${GTK_CFLAGS} ${GLIB_CFLAGS} ${MOWGLI_CFLAGS} ${ARCH_DEFINES} ${SIMD_CFLAGS} -I../..
 LDFLAGS += ${AUDLDFLAGS}
-LIBS += ${GTK_LIBS} ${GLIB_LIBS} ${MOWGLI_LIBS} -laudtag -lm
+LIBS += ${GTK_LIBS} ${GLIB_LIBS} ${MOWGLI_LIBS} -laudtag -lm -lmpg123
diff -up audacious-plugins-2.4-rc2/src/mpg123/mpg123.c~ audacious-plugins-2.4-rc2/src/mpg123/mpg123.c
--- audacious-plugins-2.4-rc2/src/mpg123/mpg123.c~	2010-08-13 23:01:12.000000000 +0200
+++ audacious-plugins-2.4-rc2/src/mpg123/mpg123.c	2010-08-24 12:00:25.904713006 +0200
@@ -33,7 +33,7 @@
 #include <audacious/plugin.h>
 #include <audacious/audtag.h>
 
-#include "libmpg123/mpg123.h"
+#include <mpg123.h>
 
 /* Define to read all frame headers when calculating file length */
 /* #define FULL_SCAN */
