diff -up audacious-plugins-2.4.4/configure~ audacious-plugins-2.4.4/configure
--- audacious-plugins-2.4.4/configure~	2011-02-22 18:29:49.000000000 +0100
+++ audacious-plugins-2.4.4/configure	2011-04-11 13:40:25.962244060 +0200
@@ -9468,20 +9468,20 @@ fi
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking for FFMPEG" >&5
 $as_echo_n "checking for FFMPEG... " >&6; }
 if test "x$PKG_CONFIG" != "x"; then
-    if { { $as_echo "$as_me:${as_lineno-$LINENO}: \$PKG_CONFIG --exists --print-errors \"libavcodec >= 52.20.0 libavcore >= 0.12.0 libavformat >= 52.31.0 libavutil >= 49.15.0\""; } >&5
-  ($PKG_CONFIG --exists --print-errors "libavcodec >= 52.20.0 libavcore >= 0.12.0 libavformat >= 52.31.0 libavutil >= 49.15.0") 2>&5
+    if { { $as_echo "$as_me:${as_lineno-$LINENO}: \$PKG_CONFIG --exists --print-errors \"libavcodec >= 52.20.0 libavformat >= 52.31.0 libavutil >= 49.15.0\""; } >&5
+  ($PKG_CONFIG --exists --print-errors "libavcodec >= 52.20.0 libavformat >= 52.31.0 libavutil >= 49.15.0") 2>&5
   ac_status=$?
   $as_echo "$as_me:${as_lineno-$LINENO}: \$? = $ac_status" >&5
   test $ac_status = 0; }; then
         if test "x${FFMPEG_CFLAGS}" != "x"; then
             pkg_cv_FFMPEG_CFLAGS="${FFMPEG_CFLAGS}"
         else
-            pkg_cv_FFMPEG_CFLAGS=`$PKG_CONFIG --cflags "libavcodec >= 52.20.0 libavcore >= 0.12.0 libavformat >= 52.31.0 libavutil >= 49.15.0" 2>/dev/null`
+            pkg_cv_FFMPEG_CFLAGS=`$PKG_CONFIG --cflags "libavcodec >= 52.20.0 libavformat >= 52.31.0 libavutil >= 49.15.0" 2>/dev/null`
         fi
         if test "x${FFMPEG_LIBS}" != "x"; then
             pkg_cv_FFMPEG_LIBS="${FFMPEG_LIBS}"
         else
-            pkg_cv_FFMPEG_LIBS=`$PKG_CONFIG --libs "libavcodec >= 52.20.0 libavcore >= 0.12.0 libavformat >= 52.31.0 libavutil >= 49.15.0" 2>/dev/null`
+            pkg_cv_FFMPEG_LIBS=`$PKG_CONFIG --libs "libavcodec >= 52.20.0 libavformat >= 52.31.0 libavutil >= 49.15.0" 2>/dev/null`
         fi
     else
         pkg_failed="yes"
@@ -9493,9 +9493,9 @@ fi
 
 if test "x$pkg_failed" = "xyes"; then
         if test "x$_pkg_short_errors_supported" = "xyes"; then
-	        FFMPEG_PKG_ERRORS=`$PKG_CONFIG --short-errors --errors-to-stdout --print-errors "libavcodec >= 52.20.0 libavcore >= 0.12.0 libavformat >= 52.31.0 libavutil >= 49.15.0" 2>&1`
+	        FFMPEG_PKG_ERRORS=`$PKG_CONFIG --short-errors --errors-to-stdout --print-errors "libavcodec >= 52.20.0 libavformat >= 52.31.0 libavutil >= 49.15.0" 2>&1`
         else
-	        FFMPEG_PKG_ERRORS=`$PKG_CONFIG --errors-to-stdout --print-errors "libavcodec >= 52.20.0 libavcore >= 0.12.0 libavformat >= 52.31.0 libavutil >= 49.15.0" 2>&1`
+	        FFMPEG_PKG_ERRORS=`$PKG_CONFIG --errors-to-stdout --print-errors "libavcodec >= 52.20.0 libavformat >= 52.31.0 libavutil >= 49.15.0" 2>&1`
         fi
 		echo "$FFMPEG_PKG_ERRORS" >&5
 	{ $as_echo "$as_me:${as_lineno-$LINENO}: result: no" >&5
