diff -up audacious-plugins-3.2/src/mpg123/mpg123.c~ audacious-plugins-3.2/src/mpg123/mpg123.c
--- audacious-plugins-3.2/src/mpg123/mpg123.c~	2012-01-20 20:48:30.000000000 +0100
+++ audacious-plugins-3.2/src/mpg123/mpg123.c	2012-01-29 21:49:43.991545176 +0100
@@ -432,7 +432,9 @@ GET_FORMAT:
 		if (! outbuf_size && (ret = mpg123_read (ctx.decoder, (void *) outbuf,
 		 sizeof outbuf, & outbuf_size)) < 0)
 		{
-			if (ret == MPG123_DONE || ret == MPG123_ERR_READER)
+			if (ret == MPG123_NEED_MORE)
+				continue;
+			if (ret == MPG123_DONE)
 				goto decode_cleanup;
 
 			fprintf (stderr, "mpg123 error in %s: %s\n", filename,
